domains/client/domain-operator/Cargo.toml

```diff
@@ -39,6 +44,7 @@ sp-trie = { git = "https://github.com/subspace/polkadot-sdk", rev = "5871818e1d7
 sp-weights = { git = "https://github.com/subspace/polkadot-sdk", rev = "5871818e1d736f1843eb9078f886290695165c42" }
 subspace-core-primitives = { version = "0.1.0", path = "../../../crates/subspace-core-primitives" }
 subspace-runtime-primitives = { version = "0.1.0", path = "../../../crates/subspace-runtime-primitives" }
+subspace-service = { version = "0.1.0", default-features = false, path = "../../../crates/subspace-service" }
```

Comment:
Ouch, this is way more problematic than the other one. This is a gigantic dependency added to a lower-level crate, which will significantly worsen compile times and will serialize compilation order even more.

Reply:
`domain-operator` hosts `domain snap sync feature` it requires both `mmr_sync` and `consensus snap sync` features from `subspace_service`. There are at least two ways to mitigate the existing dependency: either create a separate crate for sync features (both `subspace_service` and `domain_operator` will depend on it) or introduce several traits and wrapping structures to pass incapsulated functions to domains (`domain_operator` contains traits which are used by `subspace_node` to pass wrapping structures to domains). Either way, it seems more complicated than keeping an existing dependency. Do you see a simpler way or do you think the new complexity is justified from a dependency graph perspective?

Reply:
I certainly expect a separate crate (or several) for sync, we'll have even more code once we upgrade our sync implementation to implement custom `SyncStrategy` after latest Substrate upgrade. But ideally operator wouldn't depend on consensus sync either, but rather on some shared crate with common primitives/utilities.